source/core/serial_solid.cc

```diff
@@ -104,27 +98,58 @@ std::vector<
   std::pair<typename Triangulation<spacedim>::active_cell_iterator,
             typename Triangulation<dim, spacedim>::active_cell_iterator>>
 SerialSolid<dim, spacedim>::map_solid_in_background_triangulation(
-  const parallel::distributed::Triangulation<spacedim> &background_tr,
-  const std::shared_ptr<Triangulation<dim, spacedim>> & solid_tr)
+  const parallel::TriangulationBase<spacedim> &background_tr)
 {
   std::vector<
     std::pair<typename Triangulation<spacedim>::active_cell_iterator,
               typename Triangulation<dim, spacedim>::active_cell_iterator>>
     mapped_solid;
 
+
+  // Gather the amount of vertices per cell in the solid cells once to have
+  // static memory
```

Comment:
Can you explain why?

Reply:
Well not having static memory allocation is more expensive? I'm not sure what requires an explanation?

Reply:
Why do you need static memory, is it for the previous vertices of the mesh before displacement?